components/ftx/browser/ftx_service.cc

```diff
@@ -0,0 +1,398 @@
+  if (status >= 200 && status <= 299) {
+    FTXJSONParser::GetQuoteStatusJSON(body, &cost, &price, &proceeds);
+  }
+  std::move(callback).Run(cost, price, proceeds);
+}
+
+bool FTXService::ExecuteConvertQuote(const std::string& quote_id,
+                                     ExecuteConvertQuoteCallback callback) {
+  auto internal_callback = base::BindOnce(&FTXService::OnExecuteConvertQuote,
+      base::Unretained(this), std::move(callback));
+  GURL url = GetOAuthURL(
+      std::string(oauth_quote_path) + "/" + quote_id + "/accept");
+  return NetworkRequest(url, "POST", "", "",
+      std::move(internal_callback), true);
+}
+
+void FTXService::OnExecuteConvertQuote(
+    ExecuteConvertQuoteCallback callback,
+    const int status, const std::string& body,
+    const std::map<std::string, std::string>& headers) {
+  std::move(callback).Run(status >= 200 && status <= 299);
+}
+
+bool FTXService::SetAccessToken(const std::string& access_token) {
+  access_token_ = access_token;
+  std::string encrypted_access_token;
+
+  if (!OSCrypt::EncryptString(access_token, &encrypted_access_token)) {
+    LOG(ERROR) << "Could not encrypt and save FTX token info";
+    return false;
+  }
+
+  std::string encoded_encrypted_access_token;
+  base::Base64Encode(encrypted_access_token, &encoded_encrypted_access_token);
+
+  PrefService* prefs = user_prefs::UserPrefs::Get(context_);
+  prefs->SetString(kFTXAccessToken, encoded_encrypted_access_token);
+
+  return true;
+}
+
+void FTXService::ClearAuth() {
+  PrefService* prefs = user_prefs::UserPrefs::Get(context_);
+  access_token_.clear();
+  prefs->ClearPref(kFTXAccessToken);
+}
+
+bool FTXService::NetworkRequest(const GURL& url,
+                                const std::string& method,
+                                const std::string& post_data,
+                                const std::string& post_data_type,
+                                URLRequestCallback callback,
+                                bool set_auth_header) {
+  auto request = std::make_unique<network::ResourceRequest>();
+  if (set_auth_header) {
+    std::string header = GetTokenHeader();
+    request->headers.SetHeader(net::HttpRequestHeaders::kAuthorization,
+         base::StringPrintf("%s", header.c_str()));
+  }
+  request->url = url;
+  request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  request->load_flags = net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE |
+                        net::LOAD_DO_NOT_SAVE_COOKIES;
+  request->method = method;
+
+  auto url_loader = network::SimpleURLLoader::Create(
+      std::move(request), GetNetworkTrafficAnnotationTag());
+  if (!post_data.empty() && !post_data_type.empty()) {
+    url_loader->AttachStringForUpload(post_data, post_data_type);
+  }
+  url_loader->SetRetryOptions(
+      kRetriesCountOnNetworkChange,
+      network::SimpleURLLoader::RetryMode::RETRY_ON_NETWORK_CHANGE);
+
+  auto iter = url_loaders_.insert(url_loaders_.begin(), std::move(url_loader));
```

Comment:
url_loaders_ should be cleared in KeyedService::Shutdown

Reply:
fixed via https://github.com/brave/brave-core/compare/6f4c503619e82a440ef7db9782288e922c37d5e4..f4d887c0baf39543331107f2e22f9b0ad4986c02